diff -ruN libcanberra-0.28/src/cache.c libcanberra-0.28.new/src/cache.c
--- libcanberra-0.28/src/cache.c	2011-02-22 17:51:00.000000000 -0800
+++ libcanberra-0.28.new/src/cache.c	2013-02-02 14:06:22.623680828 -0800
@@ -118,7 +118,11 @@
 
         /* First we try the D-Bus machine id */
 
-        if ((f = fopen(CA_MACHINE_ID, "r"))) {
+        if ((getenv("MACHINE_ID_FILE") &&
+             (f = fopen(getenv("MACHINE_ID_FILE"), "r"))) ||
+            (f = fopen(CA_MACHINE_ID, "r")) ||
+            (f = fopen("/var/lib/dbus/machine-id", "r")) ||
+            (f = fopen("/etc/machine-id", "r"))) {
                 char ln[34] = "", *r;
 
                 r = fgets(ln, sizeof(ln)-1, f);
