diff -ruN gtk+2.0-2.24.10/gtk/gtkmodules.c gtk+2.0-2.24.10.new/gtk/gtkmodules.c
--- gtk+2.0-2.24.10/gtk/gtkmodules.c	2013-02-03 21:41:45.000000000 -0800
+++ gtk+2.0-2.24.10.new/gtk/gtkmodules.c	2013-02-03 21:50:15.945532730 -0800
@@ -54,6 +54,7 @@
 static gchar **
 get_module_path (void)
 {
+  const gchar *steam_runtime;
   const gchar *module_path_env;
   const gchar *exe_prefix;
   const gchar *home_dir;
@@ -66,34 +67,50 @@
   if (result)
     return result;
 
-  home_dir = g_get_home_dir();
-  if (home_dir)
-    home_gtk_dir = g_build_filename (home_dir, ".gtk-2.0", NULL);
-
-  module_path_env = g_getenv ("GTK_PATH");
-  exe_prefix = g_getenv ("GTK_EXE_PREFIX");
-
-  if (exe_prefix)
-    default_dir = g_build_filename (exe_prefix, "lib", "gtk-2.0", NULL);
-  else {
-    default_dir = g_build_filename (GTK_LIBDIR, "gtk-2.0", NULL);
-    pre_multiarch_dir = "/usr/lib/gtk-2.0";
-  }
-
-  if (module_path_env && home_gtk_dir)
-    module_path = g_build_path (G_SEARCHPATH_SEPARATOR_S,
-				module_path_env, home_gtk_dir, default_dir, pre_multiarch_dir, NULL);
-  else if (module_path_env)
-    module_path = g_build_path (G_SEARCHPATH_SEPARATOR_S,
-				module_path_env, default_dir, pre_multiarch_dir, NULL);
-  else if (home_gtk_dir)
-    module_path = g_build_path (G_SEARCHPATH_SEPARATOR_S,
-				home_gtk_dir, default_dir, pre_multiarch_dir, NULL);
+  steam_runtime = g_getenv ("STEAM_RUNTIME");
+  if (steam_runtime && *steam_runtime == '/')
+    {
+#if defined(__i386__)
+      const gchar *arch = "i386";
+#elif defined(__x86_64__)
+      const gchar *arch = "amd64";
+#else
+#error Unknown architecture
+#endif
+      default_dir = g_build_filename (steam_runtime, arch, GTK_LIBDIR, "gtk-2.0", NULL);
+      module_path = g_build_path (G_SEARCHPATH_SEPARATOR_S, default_dir, NULL);
+    }
   else
-    module_path = g_build_path (G_SEARCHPATH_SEPARATOR_S,
-				default_dir, pre_multiarch_dir, NULL);
+    {
+      home_dir = g_get_home_dir();
+      if (home_dir)
+        home_gtk_dir = g_build_filename (home_dir, ".gtk-2.0", NULL);
+
+      module_path_env = g_getenv ("GTK_PATH");
+      exe_prefix = g_getenv ("GTK_EXE_PREFIX");
+
+      if (exe_prefix)
+        default_dir = g_build_filename (exe_prefix, "lib", "gtk-2.0", NULL);
+      else {
+        default_dir = g_build_filename (GTK_LIBDIR, "gtk-2.0", NULL);
+        pre_multiarch_dir = "/usr/lib/gtk-2.0";
+      }
+
+      if (module_path_env && home_gtk_dir)
+        module_path = g_build_path (G_SEARCHPATH_SEPARATOR_S,
+				    module_path_env, home_gtk_dir, default_dir, pre_multiarch_dir, NULL);
+      else if (module_path_env)
+        module_path = g_build_path (G_SEARCHPATH_SEPARATOR_S,
+				    module_path_env, default_dir, pre_multiarch_dir, NULL);
+      else if (home_gtk_dir)
+        module_path = g_build_path (G_SEARCHPATH_SEPARATOR_S,
+				    home_gtk_dir, default_dir, pre_multiarch_dir, NULL);
+      else
+        module_path = g_build_path (G_SEARCHPATH_SEPARATOR_S,
+				    default_dir, pre_multiarch_dir, NULL);
 
-  g_free (home_gtk_dir);
+      g_free (home_gtk_dir);
+    }
   g_free (default_dir);
 
   result = pango_split_file_list (module_path);
