diff -ruN zenity-3.4.0/configure zenity-3.4.0.new/configure
--- zenity-3.4.0/configure	2012-03-24 19:15:42.000000000 -0700
+++ zenity-3.4.0.new/configure	2013-02-04 00:07:50.965666556 -0800
@@ -3816,7 +3816,8 @@
 
 
 
-GTK_REQUIRED=3.0.0
+GTK_REQUIRED=2.24.0
+
 
 
 
@@ -3946,12 +3947,13 @@
     pkg_cv_ZENITY_CFLAGS="$ZENITY_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gtk+-3.0 >= \$GTK_REQUIRED glib-2.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "gtk+-3.0 >= $GTK_REQUIRED glib-2.0") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gtk+-2.0 >= \$GTK_REQUIRED glib-2.0\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "gtk+-2.0 >= $GTK_REQUIRED glib-2.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_ZENITY_CFLAGS=`$PKG_CONFIG --cflags "gtk+-3.0 >= $GTK_REQUIRED glib-2.0" 2>/dev/null`
+  pkg_cv_ZENITY_CFLAGS=`$PKG_CONFIG --cflags "gtk+-2.0 >= $GTK_REQUIRED glib-2.0" 2>/dev/null`
+		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
 fi
@@ -3962,12 +3964,13 @@
     pkg_cv_ZENITY_LIBS="$ZENITY_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gtk+-3.0 >= \$GTK_REQUIRED glib-2.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "gtk+-3.0 >= $GTK_REQUIRED glib-2.0") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gtk+-2.0 >= \$GTK_REQUIRED glib-2.0\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "gtk+-2.0 >= $GTK_REQUIRED glib-2.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_ZENITY_LIBS=`$PKG_CONFIG --libs "gtk+-3.0 >= $GTK_REQUIRED glib-2.0" 2>/dev/null`
+  pkg_cv_ZENITY_LIBS=`$PKG_CONFIG --libs "gtk+-2.0 >= $GTK_REQUIRED glib-2.0" 2>/dev/null`
+		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
 fi
@@ -3987,14 +3990,14 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        ZENITY_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "gtk+-3.0 >= $GTK_REQUIRED glib-2.0" 2>&1`
+	        ZENITY_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "gtk+-2.0 >= $GTK_REQUIRED glib-2.0" 2>&1`
         else
-	        ZENITY_PKG_ERRORS=`$PKG_CONFIG --print-errors "gtk+-3.0 >= $GTK_REQUIRED glib-2.0" 2>&1`
+	        ZENITY_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "gtk+-2.0 >= $GTK_REQUIRED glib-2.0" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$ZENITY_PKG_ERRORS" >&5
 
-	as_fn_error $? "Package requirements (gtk+-3.0 >= $GTK_REQUIRED glib-2.0) were not met:
+	as_fn_error $? "Package requirements (gtk+-2.0 >= $GTK_REQUIRED glib-2.0) were not met:
 
 $ZENITY_PKG_ERRORS
 
@@ -4004,7 +4007,6 @@
 Alternatively, you may set the environment variables ZENITY_CFLAGS
 and ZENITY_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details." "$LINENO" 5
-
 elif test $pkg_failed = untried; then
      	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
@@ -4020,7 +4022,6 @@
 
 To get pkg-config, see <http://pkg-config.freedesktop.org/>.
 See \`config.log' for more details" "$LINENO" 5; }
-
 else
 	ZENITY_CFLAGS=$pkg_cv_ZENITY_CFLAGS
 	ZENITY_LIBS=$pkg_cv_ZENITY_LIBS
@@ -4034,7 +4035,7 @@
 
 zenity_save_cflags="$CFLAGS"
 # pull in the CFLAGS to locate gdkconfig.h
-CFLAGS=`$PKG_CONFIG --cflags gtk+-3.0`
+CFLAGS=`$PKG_CONFIG --cflags gtk+-2.0`
 
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -4074,6 +4075,7 @@
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
   pkg_cv_X_CFLAGS=`$PKG_CONFIG --cflags "x11" 2>/dev/null`
+		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
 fi
@@ -4090,6 +4092,7 @@
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
   pkg_cv_X_LIBS=`$PKG_CONFIG --libs "x11" 2>/dev/null`
+		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
 fi
@@ -4109,9 +4112,9 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        X_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "x11" 2>&1`
+	        X_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "x11" 2>&1`
         else
-	        X_PKG_ERRORS=`$PKG_CONFIG --print-errors "x11" 2>&1`
+	        X_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "x11" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$X_PKG_ERRORS" >&5
@@ -4126,7 +4129,6 @@
 Alternatively, you may set the environment variables X_CFLAGS
 and X_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details." "$LINENO" 5
-
 elif test $pkg_failed = untried; then
      	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
@@ -4142,7 +4144,6 @@
 
 To get pkg-config, see <http://pkg-config.freedesktop.org/>.
 See \`config.log' for more details" "$LINENO" 5; }
-
 else
 	X_CFLAGS=$pkg_cv_X_CFLAGS
 	X_LIBS=$pkg_cv_X_LIBS
@@ -4229,6 +4230,7 @@
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
   pkg_cv_LIBNOTIFY_CFLAGS=`$PKG_CONFIG --cflags "libnotify >= $LIBNOTIFY_REQUIRED" 2>/dev/null`
+		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
 fi
@@ -4245,6 +4247,7 @@
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
   pkg_cv_LIBNOTIFY_LIBS=`$PKG_CONFIG --libs "libnotify >= $LIBNOTIFY_REQUIRED" 2>/dev/null`
+		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
 fi
@@ -4264,9 +4267,9 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        LIBNOTIFY_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "libnotify >= $LIBNOTIFY_REQUIRED" 2>&1`
+	        LIBNOTIFY_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "libnotify >= $LIBNOTIFY_REQUIRED" 2>&1`
         else
-	        LIBNOTIFY_PKG_ERRORS=`$PKG_CONFIG --print-errors "libnotify >= $LIBNOTIFY_REQUIRED" 2>&1`
+	        LIBNOTIFY_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "libnotify >= $LIBNOTIFY_REQUIRED" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$LIBNOTIFY_PKG_ERRORS" >&5
@@ -4322,6 +4325,7 @@
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
   pkg_cv_WEBKIT_CFLAGS=`$PKG_CONFIG --cflags "webkitgtk-3.0 >= $WEBKITGTK_REQUIRED" 2>/dev/null`
+		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
 fi
@@ -4338,6 +4342,7 @@
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
   pkg_cv_WEBKIT_LIBS=`$PKG_CONFIG --libs "webkitgtk-3.0 >= $WEBKITGTK_REQUIRED" 2>/dev/null`
+		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
 fi
@@ -4357,9 +4362,9 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        WEBKIT_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "webkitgtk-3.0 >= $WEBKITGTK_REQUIRED" 2>&1`
+	        WEBKIT_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "webkitgtk-3.0 >= $WEBKITGTK_REQUIRED" 2>&1`
         else
-	        WEBKIT_PKG_ERRORS=`$PKG_CONFIG --print-errors "webkitgtk-3.0 >= $WEBKITGTK_REQUIRED" 2>&1`
+	        WEBKIT_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "webkitgtk-3.0 >= $WEBKITGTK_REQUIRED" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$WEBKIT_PKG_ERRORS" >&5
diff -ruN zenity-3.4.0/configure.ac zenity-3.4.0.new/configure.ac
--- zenity-3.4.0/configure.ac	2012-03-24 19:15:27.000000000 -0700
+++ zenity-3.4.0.new/configure.ac	2013-02-03 23:57:27.785656454 -0800
@@ -14,9 +14,9 @@
 # Check for programs
 AC_PROG_CC
 
-GTK_REQUIRED=3.0.0
+GTK_REQUIRED=2.24.0
 
-PKG_CHECK_MODULES([ZENITY],[gtk+-3.0 >= $GTK_REQUIRED glib-2.0])
+PKG_CHECK_MODULES([ZENITY],[gtk+-2.0 >= $GTK_REQUIRED glib-2.0])
 AC_SUBST([ZENITY_CFLAGS])
 AC_SUBST([ZENITY_LIBS])
 
@@ -26,7 +26,7 @@
 
 zenity_save_cflags="$CFLAGS"
 # pull in the CFLAGS to locate gdkconfig.h
-CFLAGS=`$PKG_CONFIG --cflags gtk+-3.0`
+CFLAGS=`$PKG_CONFIG --cflags gtk+-2.0`
 AC_COMPILE_IFELSE([
 #include <gdk/gdk.h>
 
