diff -ruN zenity-3.4.0/debian/rules zenity-3.4.0.new/debian/rules
--- zenity-3.4.0/debian/rules	2013-02-04 00:09:03.441667732 -0800
+++ zenity-3.4.0.new/debian/rules	2012-03-28 21:08:06.000000000 -0700
@@ -6,6 +6,6 @@
 include /usr/share/gnome-pkg-tools/1/rules/uploaders.mk
 -include /usr/share/gnome-pkg-tools/1/rules/gnome-get-source.mk
 
-DEB_CONFIGURE_EXTRA_FLAGS += --disable-scrollkeeper
+DEB_CONFIGURE_EXTRA_FLAGS += --disable-scrollkeeper --disable-webkitgtk
 
 LDFLAGS += -Wl,-O1 -Wl,-z,defs -Wl,--as-needed
