diff -ruN mesa-8.0.4/configs/linux-dri mesa-8.0.4.new/configs/linux-dri
--- mesa-8.0.4/configs/linux-dri	2012-03-22 10:10:07.000000000 -0700
+++ mesa-8.0.4.new/configs/linux-dri	2013-02-20 14:04:10.564825181 -0800
@@ -67,8 +67,8 @@
 INTEL_LIBS = $(shell $(PKG_CONFIG) --libs libdrm_intel)
 INTEL_CFLAGS = $(shell $(PKG_CONFIG) --cflags libdrm_intel)
 
-NOUVEAU_LIBS = $(shell $(PKG_CONFIG) --libs libdrm_nouveau)
-NOUVEAU_CFLAGS = $(shell $(PKG_CONFIG) --cflags libdrm_nouveau)
+NOUVEAU_LIBS = $(shell $(PKG_CONFIG) --libs libdrm_nouveau1)
+NOUVEAU_CFLAGS = $(shell $(PKG_CONFIG) --cflags libdrm_nouveau1)
 
 RADEON_LIBS = $(shell $(PKG_CONFIG) --libs libdrm_radeon)
 RADEON_CFLAGS = $(shell $(PKG_CONFIG) --cflags libdrm_radeon)
diff -ruN mesa-8.0.4/configure.ac mesa-8.0.4.new/configure.ac
--- mesa-8.0.4/configure.ac	2012-05-17 21:27:14.000000000 -0700
+++ mesa-8.0.4.new/configure.ac	2013-02-20 14:04:10.564825181 -0800
@@ -1284,7 +1284,7 @@
 
 case $DRI_DIRS in
 *nouveau*)
-    PKG_CHECK_MODULES([NOUVEAU], [libdrm_nouveau >= $LIBDRM_NOUVEAU_REQUIRED])
+    PKG_CHECK_MODULES([NOUVEAU], [libdrm_nouveau1 >= $LIBDRM_NOUVEAU_REQUIRED])
     ;;
 esac
 
@@ -1867,7 +1867,7 @@
             gallium_check_st "radeon/drm" "dri-r600" "xorg-r600" "" "xvmc-r600" "vdpau-r600" "va-r600"
             ;;
         xnouveau)
-            PKG_CHECK_MODULES([NOUVEAU], [libdrm_nouveau >= $LIBDRM_NOUVEAU_REQUIRED])
+            PKG_CHECK_MODULES([NOUVEAU], [libdrm_nouveau1 >= $LIBDRM_NOUVEAU_REQUIRED])
             GALLIUM_DRIVERS_DIRS="$GALLIUM_DRIVERS_DIRS nouveau nvfx nv50 nvc0"
             gallium_check_st "nouveau/drm" "dri-nouveau" "xorg-nouveau" "" "xvmc-nouveau" "vdpau-nouveau"
             ;;
diff -ruN mesa-8.0.4/src/gallium/drivers/nouveau/Makefile mesa-8.0.4.new/src/gallium/drivers/nouveau/Makefile
--- mesa-8.0.4/src/gallium/drivers/nouveau/Makefile	2012-03-22 10:10:07.000000000 -0700
+++ mesa-8.0.4.new/src/gallium/drivers/nouveau/Makefile	2013-02-20 14:04:10.564825181 -0800
@@ -7,8 +7,8 @@
 	$(LIBDRM_CFLAGS) \
 	-I$(TOP)/src/gallium/drivers/nouveau/include
 
-LIBRARY_INCLUDES += $(shell $(PKG_CONFIG) libdrm libdrm_nouveau --cflags-only-I)
-LIBRARY_DEFINES += $(shell $(PKG_CONFIG) libdrm libdrm_nouveau --cflags-only-other)
+LIBRARY_INCLUDES += $(shell $(PKG_CONFIG) libdrm libdrm_nouveau1 --cflags-only-I)
+LIBRARY_DEFINES += $(shell $(PKG_CONFIG) libdrm libdrm_nouveau1 --cflags-only-other)
 
 # get C_SOURCES
 include Makefile.sources
diff -ruN mesa-8.0.4/src/gallium/targets/dri-nouveau/Makefile mesa-8.0.4.new/src/gallium/targets/dri-nouveau/Makefile
--- mesa-8.0.4/src/gallium/targets/dri-nouveau/Makefile	2012-03-22 10:10:07.000000000 -0700
+++ mesa-8.0.4.new/src/gallium/targets/dri-nouveau/Makefile	2013-02-20 14:04:10.564825181 -0800
@@ -23,6 +23,6 @@
 
 include ../Makefile.dri
 
-DRI_LIB_DEPS += $(shell $(PKG_CONFIG) libdrm_nouveau --libs)
+DRI_LIB_DEPS += $(shell $(PKG_CONFIG) libdrm_nouveau1 --libs)
 
 symlinks:
diff -ruN mesa-8.0.4/src/gallium/targets/egl-static/Makefile mesa-8.0.4.new/src/gallium/targets/egl-static/Makefile
--- mesa-8.0.4/src/gallium/targets/egl-static/Makefile	2012-03-22 10:10:07.000000000 -0700
+++ mesa-8.0.4.new/src/gallium/targets/egl-static/Makefile	2013-02-20 14:05:20.208826312 -0800
@@ -105,7 +105,7 @@
 	$(TOP)/src/gallium/drivers/nv50/libnv50.a \
 	$(TOP)/src/gallium/drivers/nvc0/libnvc0.a \
 	$(TOP)/src/gallium/drivers/nouveau/libnouveau.a
-egl_SYS += -ldrm_nouveau
+egl_SYS += -ldrm_nouveau1
 endif
 
 # r300
diff -ruN mesa-8.0.4/src/gallium/targets/gbm/Makefile mesa-8.0.4.new/src/gallium/targets/gbm/Makefile
--- mesa-8.0.4/src/gallium/targets/gbm/Makefile	2012-03-22 10:10:07.000000000 -0700
+++ mesa-8.0.4.new/src/gallium/targets/gbm/Makefile	2013-02-20 14:05:24.680826383 -0800
@@ -66,7 +66,7 @@
 	$(TOP)/src/gallium/drivers/nv50/libnv50.a \
 	$(TOP)/src/gallium/drivers/nvc0/libnvc0.a \
 	$(TOP)/src/gallium/drivers/nouveau/libnouveau.a
-nouveau_SYS = -ldrm_nouveau
+nouveau_SYS = -ldrm_nouveau1
 
 # r300 pipe driver
 r300_LIBS = \
diff -ruN mesa-8.0.4/src/gallium/targets/vdpau-nouveau/Makefile mesa-8.0.4.new/src/gallium/targets/vdpau-nouveau/Makefile
--- mesa-8.0.4/src/gallium/targets/vdpau-nouveau/Makefile	2012-03-22 10:10:07.000000000 -0700
+++ mesa-8.0.4.new/src/gallium/targets/vdpau-nouveau/Makefile	2013-02-20 14:04:10.564825181 -0800
@@ -19,7 +19,7 @@
 	$(COMMON_GALLIUM_SOURCES) \
 	$(DRIVER_SOURCES)
 
-DRIVER_LIBS = $(shell $(PKG_CONFIG) libdrm_nouveau --libs) -lXfixes
+DRIVER_LIBS = $(shell $(PKG_CONFIG) libdrm_nouveau1 --libs) -lXfixes
 
 LINK_WITH_CXX=1
 
diff -ruN mesa-8.0.4/src/gallium/targets/xorg-nouveau/Makefile mesa-8.0.4.new/src/gallium/targets/xorg-nouveau/Makefile
--- mesa-8.0.4/src/gallium/targets/xorg-nouveau/Makefile	2012-03-22 10:10:07.000000000 -0700
+++ mesa-8.0.4.new/src/gallium/targets/xorg-nouveau/Makefile	2013-02-20 14:04:10.564825181 -0800
@@ -21,10 +21,10 @@
 	$(TOP)/src/gallium/drivers/rbug/librbug.a
 
 DRIVER_LINKS = \
-	$(shell $(PKG_CONFIG) --libs libdrm libdrm_nouveau)
+	$(shell $(PKG_CONFIG) --libs libdrm libdrm_nouveau1)
 
 DRIVER_INCLUDES = \
-	$(shell $(PKG_CONFIG) --cflags-only-I libdrm libdrm_nouveau xf86driproto)
+	$(shell $(PKG_CONFIG) --cflags-only-I libdrm libdrm_nouveau1 xf86driproto)
 
 LINK_WITH_CXX=1
 
diff -ruN mesa-8.0.4/src/gallium/targets/xvmc-nouveau/Makefile mesa-8.0.4.new/src/gallium/targets/xvmc-nouveau/Makefile
--- mesa-8.0.4/src/gallium/targets/xvmc-nouveau/Makefile	2012-03-22 10:10:07.000000000 -0700
+++ mesa-8.0.4.new/src/gallium/targets/xvmc-nouveau/Makefile	2013-02-20 14:04:10.564825181 -0800
@@ -19,7 +19,7 @@
 	$(COMMON_GALLIUM_SOURCES) \
 	$(DRIVER_SOURCES)
 
-DRIVER_LIBS = $(shell $(PKG_CONFIG) libdrm_nouveau --libs) -lXfixes
+DRIVER_LIBS = $(shell $(PKG_CONFIG) libdrm_nouveau1 --libs) -lXfixes
 
 LINK_WITH_CXX=1
 
diff -ruN mesa-8.0.4/src/gallium/winsys/nouveau/drm/Makefile mesa-8.0.4.new/src/gallium/winsys/nouveau/drm/Makefile
--- mesa-8.0.4/src/gallium/winsys/nouveau/drm/Makefile	2012-03-22 10:10:07.000000000 -0700
+++ mesa-8.0.4.new/src/gallium/winsys/nouveau/drm/Makefile	2013-02-20 14:04:10.564825181 -0800
@@ -5,7 +5,7 @@
 
 C_SOURCES = nouveau_drm_winsys.c
 
-LIBRARY_INCLUDES = $(shell $(PKG_CONFIG) libdrm libdrm_nouveau --cflags-only-I)
-LIBRARY_DEFINES = $(shell $(PKG_CONFIG) libdrm libdrm_nouveau --cflags-only-other)
+LIBRARY_INCLUDES = $(shell $(PKG_CONFIG) libdrm libdrm_nouveau1 --cflags-only-I)
+LIBRARY_DEFINES = $(shell $(PKG_CONFIG) libdrm libdrm_nouveau1 --cflags-only-other)
 
 include ../../../Makefile.template
