diff -ruN gdk-pixbuf-2.26.1/gdk-pixbuf/gdk-pixbuf-io.c gdk-pixbuf-2.26.1.new/gdk-pixbuf/gdk-pixbuf-io.c
--- gdk-pixbuf-2.26.1/gdk-pixbuf/gdk-pixbuf-io.c	2012-02-04 15:21:30.000000000 -0800
+++ gdk-pixbuf-2.26.1.new/gdk-pixbuf/gdk-pixbuf-io.c	2013-02-04 01:32:20.765748744 -0800
@@ -389,10 +389,29 @@
 static gchar *
 gdk_pixbuf_get_module_file (void)
 {
-  gchar *result = g_strdup (g_getenv ("GDK_PIXBUF_MODULE_FILE"));
+  const gchar *steam_runtime;
+  gchar *result;
 
-  if (!result)
+  /* First load the steam runtime, if available */
+  steam_runtime = g_getenv ("STEAM_RUNTIME");
+  if (steam_runtime && *steam_runtime == '/')
+    {
+#if defined(__i386__)
+      const gchar *arch = "i386";
+#elif defined(__x86_64__)
+      const gchar *arch = "amd64";
+#else
+#error Unknown architecture
+#endif
+      result = g_build_filename (steam_runtime, arch, GDK_PIXBUF_LIBDIR, "gdk-pixbuf-2.0", GDK_PIXBUF_BINARY_VERSION, "loaders.cache", NULL);
+    }
+  else
+    {
+      result = g_strdup (g_getenv ("GDK_PIXBUF_MODULE_FILE"));
+
+      if (!result)
           result = g_build_filename (GDK_PIXBUF_LIBDIR, "gdk-pixbuf-2.0", GDK_PIXBUF_BINARY_VERSION, "loaders.cache", NULL);
+    }
 
   return result;
 }
