diff -ruN alsa-lib-1.0.25/src/dlmisc.c alsa-lib-1.0.25.new/src/dlmisc.c
--- alsa-lib-1.0.25/src/dlmisc.c	2013-02-17 22:30:16.000000000 -0800
+++ alsa-lib-1.0.25.new/src/dlmisc.c	2013-02-17 22:30:03.616292103 -0800
@@ -75,10 +75,28 @@
 	void *handle = NULL;
 	char *filename;
 
-	if (name && name[0] != '/') {
-		filename = malloc(sizeof(ALSA_PLUGIN_DIR) + 1 + strlen(name) + 1);
-		strcpy(filename, ALSA_PLUGIN_DIR);
-		strcat(filename, "/");
+	if (name) {
+		const char *steam_runtime = getenv("STEAM_RUNTIME");
+		if (steam_runtime && *steam_runtime == '/') {
+#if defined(__i386__)
+			const char *arch = "i386";
+#elif defined(__x86_64__)
+			const char *arch = "amd64";
+#else
+#error Unknown architecture
+#endif
+			filename = malloc(strlen(steam_runtime) + 1 + strlen(arch) + 1 + sizeof(ALSA_PLUGIN_DIR) + 1 + strlen(name) + 1);
+			strcpy(filename, steam_runtime);
+			strcat(filename, "/");
+			strcat(filename, arch);
+		} else {
+			filename = malloc(sizeof(ALSA_PLUGIN_DIR) + 1 + strlen(name) + 1);
+			*filename = '\0';
+		}
+		if (*name != '/') {
+			strcat(filename, ALSA_PLUGIN_DIR);
+			strcat(filename, "/");
+		}
 		strcat(filename, name);
 		handle = dlopen(filename, mode);
 		free(filename);
