diff -ruN libsdl2-mixer-2.0.0/debian/rules libsdl2-mixer-2.0.0.new/debian/rules
--- libsdl2-mixer-2.0.0/debian/rules	2013-06-28 09:46:34.000000000 -0700
+++ libsdl2-mixer-2.0.0.new/debian/rules	2013-06-28 09:48:21.521706056 -0700
@@ -26,20 +26,6 @@
 confflags += --enable-music-midi-fluidsynth
 confflags += --enable-music-midi-timidity
 
-# mafm 20111203: removed the "disable shared" support from previous versions, I
-# couldn't find any reason for it in the changelog
-#
-# mafm 20111204: issue "disable-*-shared" flags for 'configure' again, otherwise
-# SDL_mixer library opens those libraries in a plugin-like way (with
-# SDL_LoadObject), and dpkg-shlibdeps doesn't seem to generate the needed
-# dependencies in the binary library package
-confflags += --disable-music-flac-shared
-confflags += --disable-music-ogg-shared
-confflags += --disable-music-mp3-smpeg-shared
-confflags += --disable-music-mod-mikmod-shared
-confflags += --disable-music-mod-modplug-shared
-confflags += --disable-music-fluidsynth-shared
-
 %:
 	dh $@ --with autoreconf --parallel
 
