diff -ruN libsdl2-image-2.0.0/debian/rules libsdl2-image-2.0.0.new/debian/rules
--- libsdl2-image-2.0.0/debian/rules	2013-02-18 08:28:06.000000000 -0800
+++ libsdl2-image-2.0.0.new/debian/rules	2013-04-01 23:59:40.028855847 -0700
@@ -1,6 +1,6 @@
 #!/usr/bin/make -f
 
-confflags = --disable-webp --disable-jpg-shared --disable-png-shared --disable-tif-shared
+confflags = --disable-webp
 
 %:
 	dh $@ --with autoreconf --parallel
