diff -ruN libsdl2-2.0.0/debian/rules libsdl2-2.0.0.new/debian/rules
--- libsdl2-2.0.0/debian/rules	2013-02-18 20:24:23.000000000 -0800
+++ libsdl2-2.0.0.new/debian/rules	2013-03-04 11:54:52.263300335 -0800
@@ -4,9 +4,7 @@
 DEB_HOST_MULTIARCH ?= $(shell dpkg-architecture -qDEB_HOST_MULTIARCH)
 
 confflags = --disable-rpath --disable-video-directfb \
-            --disable-nas --disable-esd --disable-arts \
-            --disable-alsa-shared --disable-pulseaudio-shared \
-            --disable-x11-shared
+            --disable-nas --disable-esd --disable-arts
 
 %:
 	dh $@ --parallel
