Removed codecs.

--- libav-0.8.5/debian/confflags	2012-01-21 00:32:52.000000000 -0800
+++ libav-0.8.5.new/debian/confflags	2013-11-01 12:07:48.000000000 -0700
@@ -97,58 +97,65 @@
 confflags += --arch='$(DEB_HOST_ARCH_CPU)'
 confflags += --prefix=/usr
 confflags += --libdir=/usr/lib/$(DEB_HOST_MULTIARCH)
-confflags += --enable-vdpau
-confflags += --enable-bzlib
-confflags += --enable-libgsm
-confflags += --enable-libschroedinger
-confflags += --enable-libspeex
-confflags += --enable-libtheora
-confflags += --enable-libvorbis
-confflags += --enable-pthreads
-confflags += --enable-zlib
-confflags += --enable-libvpx
-confflags += --enable-runtime-cpudetect
-confflags += --enable-libfreetype
-ifeq      ($(DEB_HOST_ARCH),armel)
-# this is required on Ubuntu lucid as it defaults to thumb2 and Libav has
-# plenty of incompatible assembly; not sure how to detect that properly
-confflags += --extra-cflags="-marm -fPIC -DPIC"
-endif
-
-ifeq      ($(DEB_HOST_ARCH),powerpc)
-confflags += --enable-pic
-endif
+# confflags += --enable-vdpau
+# confflags += --enable-bzlib
+# confflags += --enable-libgsm
+# confflags += --enable-libschroedinger
+# confflags += --enable-libspeex
+# confflags += --enable-libtheora
+# confflags += --enable-libvorbis
+# confflags += --enable-pthreads
+# confflags += --enable-zlib
+# confflags += --enable-libvpx
+# confflags += --enable-runtime-cpudetect
+# confflags += --enable-libfreetype
+# ifeq      ($(DEB_HOST_ARCH),armel)
+# # this is required on Ubuntu lucid as it defaults to thumb2 and Libav has
+# # plenty of incompatible assembly; not sure how to detect that properly
+# confflags += --extra-cflags="-marm -fPIC -DPIC"
+# endif
+
+# ifeq      ($(DEB_HOST_ARCH),powerpc)
+# confflags += --enable-pic
+# endif
 
-confflags += $(extra_common_confflags)
+# confflags += $(extra_common_confflags)
 
 # vaapi is not available on s390 and on the hurd
-confflags += $(call cond_enable,/usr/include/va/va.h,vaapi)
+# confflags += $(call cond_enable,/usr/include/va/va.h,vaapi)
 
 # this part below is intended for the 'Libav' package in ubuntu/multiverse
-gpl_confflags += $(call cond_enable,/usr/include/dirac/libdirac_decoder/dirac_parser.h,libdirac)
-gpl_confflags += $(call cond_enable,/usr/include/lame/lame.h,libmp3lame)
-gpl_confflags += $(call cond_enable,/usr/include/librtmp/http.h,librtmp)
-gpl_confflags += $(call cond_enable,/usr/include/x264.h,libx264)
-gpl_confflags += $(call cond_enable,/usr/include/xvid.h,libxvid)
+# gpl_confflags += $(call cond_enable,/usr/include/dirac/libdirac_decoder/dirac_parser.h,libdirac)
+# gpl_confflags += $(call cond_enable,/usr/include/lame/lame.h,libmp3lame)
+# gpl_confflags += $(call cond_enable,/usr/include/librtmp/http.h,librtmp)
+# gpl_confflags += $(call cond_enable,/usr/include/x264.h,libx264)
+# gpl_confflags += $(call cond_enable,/usr/include/xvid.h,libxvid)
 
-confflags += $(call cond_enable,/usr/include/openjpeg.h,libopenjpeg)
+# confflags += $(call cond_enable,/usr/include/openjpeg.h,libopenjpeg)
 
 # Opencore-amr requires GPL v3
-v3_confflags += $(call cond_enable_v3,/usr/include/opencore-amrnb/interf_dec.h,libopencore-amrnb)
-v3_confflags += $(call cond_enable_v3,/usr/include/opencore-amrwb/dec_if.h,libopencore-amrwb)
+# v3_confflags += $(call cond_enable_v3,/usr/include/opencore-amrnb/interf_dec.h,libopencore-amrnb)
+# v3_confflags += $(call cond_enable_v3,/usr/include/opencore-amrwb/dec_if.h,libopencore-amrwb)
 
 # AAC is considered non-free upstream
-confflags += $(call cond_enable_nf,/usr/include/faac.h,libfaac)
+# confflags += $(call cond_enable_nf,/usr/include/faac.h,libfaac)
 
 # comment out following line for LGPL versions of the libraries
 confflags += $(gpl_confflags)
-confflags += $(v3_confflags)
+# confflags += $(v3_confflags)
+
+# --start
+#confflags += --disable-encoders
+#confflags += --disable-decoders
+#confflags += --disable-filters
+confflags += --disable-everything
+# --end
 
 # Enable IEEE 1394 (FireWire) support on Linux only
-ifneq (,$(findstring linux,$(DEB_HOST_GNU_TYPE)))
-  confflags += --enable-libdc1394
-  lib1394-dev += libraw1394-dev, libdc1394-22-dev
-endif
+# ifneq (,$(findstring linux,$(DEB_HOST_GNU_TYPE)))
+#   confflags += --enable-libdc1394
+#   lib1394-dev += libraw1394-dev, libdc1394-22-dev
+# endif
 
 # Sanitize injected compiler in linker flags
 CFLAGS :=
