diff -ruN pango1.0-1.30.0/debian/dh_pangomodules.in pango1.0-1.30.0.new/debian/dh_pangomodules.in
--- pango1.0-1.30.0/debian/dh_pangomodules.in	2012-06-14 15:16:05.000000000 -0700
+++ pango1.0-1.30.0.new/debian/dh_pangomodules.in	2013-01-28 18:22:38.728768966 -0800
@@ -139,8 +139,8 @@
         chomp;
         next if /^$/;
         # strip build-time prefix from output
-        if (m#^\Q$absolute_tmp/\E#) {
-            s#^\Q$absolute_tmp/\E#/#;
+        if (m#.*/modules/\E#) {
+            s#.*/modules/\E#modules/#;
             complex_doit("printf '%s\\n' '$_' >>$module_file");
             next;
         }
