diff -ruN pango1.0-1.30.0/pango/pango-utils.c pango1.0-1.30.0.new/pango/pango-utils.c
--- pango1.0-1.30.0/pango/pango-utils.c	2011-08-15 18:11:08.000000000 -0700
+++ pango1.0-1.30.0.new/pango/pango-utils.c	2013-02-03 21:10:06.585493669 -0800
@@ -741,11 +741,26 @@
 	result = g_build_filename (root, "lib\\pango", NULL);
       g_free (root);
 #else
-      const char *libdir = g_getenv ("PANGO_LIBDIR");
-      if (libdir != NULL)
-	result = g_build_filename (libdir, "pango", NULL);
+      const gchar *steam_runtime = g_getenv ("STEAM_RUNTIME");
+      if (steam_runtime && *steam_runtime == '/')
+        {
+#if defined(__i386__)
+          const gchar *arch = "i386";
+#elif defined(__x86_64__)
+          const gchar *arch = "amd64";
+#else
+#error Unknown architecture
+#endif
+	  result = g_build_filename (steam_runtime, arch, LIBDIR, "pango", NULL);
+        }
       else
-	result = LIBDIR "/pango";
+        {
+          const char *libdir = g_getenv ("PANGO_LIBDIR");
+          if (libdir != NULL)
+	    result = g_build_filename (libdir, "pango", NULL);
+          else
+	    result = LIBDIR "/pango";
+        }
 #endif
     }
   return result;
