diff -ruN pulseaudio-1.1/src/pulsecore/core-util.c pulseaudio-1.1.new/src/pulsecore/core-util.c
--- pulseaudio-1.1/src/pulsecore/core-util.c	2011-10-20 05:54:16.000000000 -0700
+++ pulseaudio-1.1.new/src/pulsecore/core-util.c	2013-02-02 13:15:22.599631219 -0800
@@ -2697,7 +2697,11 @@
      * since it fits perfectly our needs and is not as volatile as the
      * hostname which might be set from dhcp. */
 
-    if ((f = pa_fopen_cloexec(PA_MACHINE_ID, "r"))) {
+    if ((getenv("MACHINE_ID_FILE") &&
+         (f = pa_fopen_cloexec(getenv("MACHINE_ID_FILE"), "r"))) ||
+        (f = pa_fopen_cloexec(PA_MACHINE_ID, "r")) ||
+        (f = pa_fopen_cloexec("/var/lib/dbus/machine-id", "r")) ||
+        (f = pa_fopen_cloexec("/etc/machine-id", "r"))) {
         char ln[34] = "", *r;
 
         r = fgets(ln, sizeof(ln)-1, f);
