diff -ruN libx11-1.4.99.1/src/xlibi18n/lcFile.c libx11-1.4.99.1.new/src/xlibi18n/lcFile.c
--- libx11-1.4.99.1/src/xlibi18n/lcFile.c	2011-10-13 20:43:40.000000000 -0700
+++ libx11-1.4.99.1.new/src/xlibi18n/lcFile.c	2013-02-06 11:07:14.529689805 -0800
@@ -226,11 +226,27 @@
     int len = 0;
 
 #ifndef NO_XLOCALEDIR
+    char steam_runtime_locale_dir[PATH_MAX];
     char *dir;
     int priv = 1;
 
     dir = getenv("XLOCALEDIR");
 
+    if (!dir) {
+#if defined(__i386__)
+        const char *arch = "i386";
+#elif defined(__x86_64__)
+        const char *arch = "amd64";
+#else
+#error Unknown architecture
+#endif
+        const char *steam_runtime = getenv("STEAM_RUNTIME");
+        if (steam_runtime && *steam_runtime == '/') {
+            snprintf(steam_runtime_locale_dir, sizeof(steam_runtime_locale_dir), "%s/%s/%s", steam_runtime, arch, XLOCALEDIR);
+            dir = steam_runtime_locale_dir;
+        }
+    }
+
     if (dir) {
 #ifndef WIN32
 	/*
@@ -304,11 +320,27 @@
     int len = 0;
 
 #ifndef NO_XLOCALEDIR
+    char steam_runtime_locale_dir[PATH_MAX];
     char *dir;
     int priv = 1;
 
     dir = getenv("XLOCALELIBDIR");
 
+    if (!dir) {
+#if defined(__i386__)
+        const char *arch = "i386";
+#elif defined(__x86_64__)
+        const char *arch = "amd64";
+#else
+#error Unknown architecture
+#endif
+        const char *steam_runtime = getenv("STEAM_RUNTIME");
+        if (steam_runtime && *steam_runtime == '/') {
+            snprintf(steam_runtime_locale_dir, sizeof(steam_runtime_locale_dir), "%s/%s/%s", steam_runtime, arch, XLOCALEDIR);
+            dir = steam_runtime_locale_dir;
+        }
+    }
+
     if (dir) {
 #ifndef WIN32
 	/*
